### PR TITLE
Huge refactor, switch to PWMOut compatible API.

### LINKED ISSUE #6: Switch to properties instead of weird functions.
This driver has the MicroPython-ism of a setter/getter function where its a getter when values are None and setter when they are not. This is not Pythonic and confusing.

Instead, this API should be changed so that it has a number of channels that act like [`pulseio.PWMOut`](https://circuitpython.readthedocs.io/en/stable/shared-bindings/pulseio/PWMOut.html) with `frequency` and `duty_cycle` as properties. `frequency` of each channel would be read only because its shared amongst all channels. The outer class can have an adjustable frequency property.

### LINKED ISSUE #7: Include all boilerplate from cookiecutter.
The [CircuitPython CookieCutter](https://github.com/adafruit/cookiecutter-adafruit-circuitpython) has a standard set of files for CircuitPython repos. This repo predates the template and needs to be updated.

### PR DESCRIPTION
**Pull request by tannewt**
This change reduces the scope of the PCA9685 driver to simply
providing PWMOut compatible objects for each channel. As a result,
the motor and servo can now be used with any PWM source, not just
the PCA9685. The new motor library is here:
https://github.com/adafruit/Adafruit_CircuitPython_Motor

Furthermore, this library is not aware of particular output configs
such as the specific Adafruit Motor FeatherWing. That knowledge has
been moved to a new FeatherWing library that handles all wing
related initialization. Its available here:
https://github.com/adafruit/Adafruit_CircuitPython_FeatherWing

Lastly, the new driver code and examples have been linted and
standard library files added. This part is related to https://github.com/adafruit/circuitpython/issues/475

These two guides will need to be updated alongside the release:
https://learn.adafruit.com/micropython-hardware-pca9685-pwm-and-servo-driver/micropython?view=all#circuitpython
https://learn.adafruit.com/micropython-hardware-pca9685-dc-motor-and-stepper-driver/micropython?view=all

Fixes #6 and fixes #7.

### CODE REVIEW

File: examples/calibration.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import time
```
Comment (kattni) [inline examples/calibration.py]: Clarify what the calibration is for.

File: examples/simpletest.py [hunk 2]
```diff
@@ -1,18 +1,18 @@
-from board import *
+from board import SCL, SDA
```
Comment (kattni) [inline examples/simpletest.py]: Clarify what `simpletest` is and how it should be used.

File: examples/calibration.py (hunk 1 above)
Comment (tannewt) [reply examples/calibration.py]: Done

File: examples/simpletest.py (hunk 2 above)
Comment (tannewt) [reply examples/simpletest.py]: Done

**Pull request merged**